docs/asm-gke-terraform/README.md

```diff
@@ -0,0 +1,118 @@
+
+### 1. Enable the APIs listed above
+```
+gcloud services enable mesh.googleapis.com
+```
+### 2.  Clone this repo and cd to this directory
+```
+git clone https://github.com/GoogleCloudPlatform/anthos-service-mesh-samples
+cd anthos-service-mesh-samples/docs/asm-gke-terraform
+```
+### 3. Set your Google Cloud Platform `PROJECT_ID`
+
+Set the `PROJECT_ID` environment variable:
+
+To enable the above services, run the following in your terminal
+```
+export PROJECT_ID="<YOUR_PROJECT_ID>"
+gcloud config set project $PROJECT_ID
+```
+### 4.  Create a `variables.tfvars` to set your project ID for Terraform
+Run the following command to create a `variables.tfvars` file
+```
+echo "project_id = \"$PROJECT_ID\"" > terraform.tfvars
+```
+
+### 5.  Deploy the Terrform module to set up your ASM on the GKE Cluster
+```
+terraform init
+terraform plan 
+terraform apply 
+```
+Enter `yes` to confirm the Terraform apply step.
+
+`terraform apply` will take a few minutes to provision on GCP
+
+## Verify Anthos Service Mesh Installation
+### 1.  Verify that your GKE Cluster membership to a `Fleet` was successful 
+A **[Fleet](https://cloud.google.com/anthos/multicluster-management/fleets)** is the term used to logically organized clusters and other resources, for easier management of multi-clusters projects. 
+```
+gcloud container fleet memberships list --project $PROJECT_ID
+```
+
+### 2. Inspect your `controlplanerevision` Custom Resource 
+Run the command to check that the Anthos Service Mesh Control Plane has been succesfully enabled in your cluster
+```
+gcloud container fleet mesh describe
+```
+The output should be similar to the following. Note that the `description` is set to `Revision(s) ready for use: asm-managed`
+```
+createTime: '2022-11-28T14:40:57.866893805Z'
+membershipSpecs:
+  projects/932729642594/locations/global/memberships/my-membership:
+    mesh:
+      management: MANAGEMENT_AUTOMATIC
+membershipStates:
+  projects/932729642594/locations/global/memberships/my-membership:
+    servicemesh:
+      controlPlaneManagement:
+        details:
+        - code: REVISION_READY
+          details: 'Ready: asm-managed'
+        state: ACTIVE
+      dataPlaneManagement:
+        details:
+        - code: OK
+          details: Service is running.
+        state: ACTIVE
+    state:
+      code: OK
+      description: 'Revision(s) ready for use: asm-managed.'
+      updateTime: '2022-11-28T15:41:18.013711333Z'
+name: projects/christineskim-asm-tf-04/locations/global/features/servicemesh
+resourceState:
+  state: ACTIVE
+spec: {}
+state:
+  state: {}
+updateTime: '2022-11-28T15:41:23.056954826Z'
+```
+### 3. Retrieve your GKE Cluster credentials and review the control plane state: 
+```
+gcloud container clusters get-credentials "asm-cluster" --region "us-central1" --project $PROJECT_ID
+```
+Look at the status of your `controlplanerevision` CRD with the following command: 
+```
+kubectl get controlplanerevisions -n istio-system
```

Comment:
Are we sure we want to illustrate/show the `controlplanrevisions` resource? Is `gcloud container fleet mesh describe` not enough for an end users perspective?

Reply:
i think it would be beneficial to see _within_ the cluster of `asm-managed`

Reply:
The `gcloud` command already provides this too, and this `gcloud` command is to abstract the access of this `controlplanerevision` resource and the cluster itself: `gcloud` doesn't need `get-credentials`, `kubectl` does :)

Reply:
hmmmm in the guide itself that will live, i believe the tws would like to have steps afterwards - deploy online boutique, so getting creds will be needed
i can take it out though, i see your point